### ISSUE TITLE
TypeScript default overloads treated as multiple default exports

### ISSUE THREAD

**Issue by AndrewLeedham** [import-js/eslint-plugin-import#1590]
The following example causes 3 `error  Multiple default exports  import/export` errors, but there is only technically 1 default export the other 2 are overloads.
```TypeScript
export default function foo(param: string): boolean;
export default function foo(param: string, param1: number): boolean;
export default function foo(param: string, param1?: number): boolean {
  return param && param1;
}
```
There seems to be some code for checking overloads: https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/export.js#L37-L40 but this does not seem to catch this case. I added some debug logs in and the rough contents of namespace is:
```
root: {
  default: [
    {
       type: 'ExportDefaultDeclaration'
    },
    {
       type: 'ExportDefaultDeclaration'
    },
    {
       type: 'ExportDefaultDeclaration'
    }
  ]
}
```

I am running this in the following environment:
* **eslint**: 5.16.0
* **eslint-plugin-import**: 2.19.1
* **eslint-import-resolver-typescript**: 2.0.0
* **typescript**: 3.7.2
* **@typescript-eslint/parser**: 2.13.0

Related:
* https://github.com/benmosher/eslint-plugin-import/issues/22#issuecomment-569955880
* https://github.com/benmosher/eslint-plugin-import/issues/1357

**Comment by benjie** [comment]
A workaround is:

```ts
function foo(param: string): boolean;
function foo(param: string, param1: number): boolean;
function foo(param: string, param1?: number): boolean {
  return param && param1;
}

export default foo;
```

**Comment by eyedean** [comment]
@benjie sure, that unblocks the case.  But, ideally, it should be able to detect it's an overload.

**Comment by AverageHelper** [comment]
Can confirm this issue is still present. I too think that the plugin should be able to detect that this case is an overload, and not actually a case of multiple default exports.

**Comment by azhaochen** [comment]
> @benjie sure, that unblocks the case. But, ideally, it should be able to detect it's an overload.
> 
> PS. Might be obvious, but just in case for beginners, for non-default exports you similarly do:
> 
> ```ts
> function foo(param: string): boolean;
> function foo(param: string, param1: number): boolean;
> function foo(param: string, param1?: number): boolean {
>   return param && param1;
> }
> 
> export { foo };
> ```

export { foo } should put in the first line, or it will report error 'An export declaration can only be used at the top level of a namespace or module.ts(1233)'

**Comment by jordanbtucker** [comment]
Just checking to see if there's an ETA for this fix to land in a release.

**Comment by ljharb** [comment]
No, there’s no timeline for releases.

**Comment by TomasHubelbauer** [comment]
```typescript
export default function test(str: string): void;
export default function test(num: number, str: string): void;
export default function test(arg: string | number, str?: string) {
  console.log(arg, str);
}
```

Output:

> Multiple default exports. eslint(import/export)

Looks like this is still present in some form or another. Or is my overload incorrect? There are no errors when pasting the above snippet to the TypeScript Playground verbatim.

**Comment by ljharb** [comment]
@TomasHubelbauer what version of the plugin are you using?

**Comment by TomasHubelbauer** [comment]
2.26.0_eslint@8.40.0 - I am using it through `eslint-config-next` 13.4.1

**Comment by ljharb** [comment]
@TomasHubelbauer the fix for this was in v2.27.5, so that's your problem.

**Comment by TomasHubelbauer** [comment]
I see! Thanks!
